### ISSUE TITLE
Any plans about speeding up?

### ISSUE THREAD

**Issue by cxxgtxy** [open-mmlab/mmselfsup#15]
Thanks for such impressive work.
However, the training settings mainly focus on a batch size of 256.  
I have tested the cost on V100 and it would cost about 7 days to train MoCo/Simclr using Res50 (default settings). 
I double batch size and learning rate, however, the training speed remains almost unchanged. Therefore, perhaps, AMP or DALI will be used to speed up the training process.

Moreover,  have you tested the results with a larger batch size? If you have done,  are the results comparable with the reported scores based on 256?
Thanks!

**Comment by XiaohangZhan** [comment]
It is really weird. I've trained MoCo with 8 GTX1080TI GPUs and it requires about 4 days. V100 should be faster than GTX1080TI. How may V100s are you using? Batch size 256 corresponds to 8 GPUs.
We have not tried a larger batch size due to the limitation of GPUs.

**Comment by cxxgtxy** [comment]
A single V100 machine with 4 GPU cards.
It seems that your 1080 (8cards) is faster than my V100 (4 cards).
However, the avg gpu utilization rate is about 75%   using the nvidia-smi tools.
Do you use memcache?

**Comment by XiaohangZhan** [comment]
Did you mean that you use 4 GPUs? Then `imgs_per_gpu` should double in your config.
I used memcache in training. It is applicable in my slurm environment. You may also set up your own memcache client.

**Comment by cxxgtxy** [comment]
I doubled imgs_per_gpu.
Thanks! I will try memcache.

**Comment by cxxgtxy** [comment]
One more question about memcache. I confirm that mc is from mmcv. However, the format of conf is not given. 
Can you show a tiny example?
        client_config_file = "{}/client.conf".format(self.mclient_path)
        self.mclient = mc.MemcachedClient.GetInstance(server_list_config_file,
                                                      client_config_file)

Thanks!

**Comment by XiaohangZhan** [comment]
4 GPUs will be slower than 8 GPUs with the same total batch size. This may be one of the causes.
You may not be able to use memcache via merely get these conf files. You have to set up the Memcached client in your slurm first (https://github.com/memcached/memcached/wiki/Install).
Besides, your slurm must contain multiple machines, otherwise, memcache will take no effect.
Below is my client.conf
```
INIT_GET_POOL_SIZE: 16 # initial size of get connection pool
MAX_GET_POOL_SIZE: 64 # maximum size of get connection pool
INIT_PUT_POOL_SIZE: 16 # initial size of put connection pool
MAX_PUT_POOL_SIZE: 64 # maximum size of put connection pool
WAIT_TIME_FOR_CONN_SEC: 0 # how long we wait for until a connection is available in second
WAIT_TIME_FOR_CONN_NANOSEC: 1000000 # how long we wait for until a connection is available in nano second
NUM_READS_FOR_DISP_HIT_RATIO: 10000 # how many number of reads we wait for a display of hit ratio
GET_POOL_CONNECT_TIMEOUT: 10 # connection timeout for get connection pool
PUT_POOL_CONNECT_TIMEOUT: 10 # connection timeout for put connection pool
GET_POOL_POLL_TIMEOUT: 20 # polling timeout for get connection pool
PUT_POOL_POLL_TIMEOUT: 20 # polling timeout for put connection pool
```

**Comment by cxxgtxy** [comment]
Thanks again.

**Comment by guochengqian** [comment]
@XiaohangZhan 
Hello. When you double the images per GPU, do you have to divide the LR by 2?
I see from the get_started.md, you mention that the LR should also be changed right?
So we have to change both of them at the same time?

**Comment by XiaohangZhan** [comment]
I guess you mean we should double the LR, too. It is just empirical, mentioned in previous papers such as SimCLR. Of course, you do not have to adjust the LR, but empirically changing LR according to this formula should be better.

**Comment by guochengqian** [comment]
Thank you! @XiaohangZhan
